### PR TITLE
Version 2

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by lapineige** [comment]
!testme

File: scripts/upgrade [hunk 1]
```diff
@@ -111,88 +32,45 @@ if [ "$upgrade_type" == "UPGRADE_APP" ]
...
 #=================================================
-# INTEGRATE SERVICE IN YUNOHOST
+# INSTALL PYTHON DEPENDENCIES
 #=================================================
-ynh_script_progression --message="Integrating service in YunoHost..." --weight=1
+ynh_script_progression --message="Installing python dependencies..." --weight=1
 
-yunohost service add $app --description="A short description of the app" --log="/var/log/$app/$app.log"
+pushd $install_dir
+	ynh_exec_as $app python3 -m venv $final_path/venv
```
Comment (alexAubin) [inline scripts/upgrade:63]:
```suggestion
	ynh_exec_as $app python3 -m venv $install_dir/venv
```

File: scripts/install [hunk 2]
```diff
@@ -9,142 +9,45 @@
...
 #=================================================
-ynh_script_progression --message="Integrating service in YunoHost..." --weight=1
+ynh_script_progression --message="Installing python dependencies..." --weight=1
 
-yunohost service add $app --description="A short description of the app" --log="/var/log/$app/$app.log"
+pushd $install_dir
+	ynh_exec_as $app python3 -m venv $install_dir/venv
+	ynh_exec_as $app "$install_dir/venv/bin/pip" install --upgrade pip
+	ynh_exec_as $app "$install_dir/venv/bin/pip" install -r requirements.txt
+popd
```
Comment (alexAubin) [inline scripts/install:42]:
```suggestion
popd

# Tmp debug ...
ls -l $install_dir
ls -l $install_dir/venv/
ls -l $install_dir/venv/bin/
ls -l $install_dir/venv/bin/python3
```

**Comment by alexAubin** [comment]
!testme

File: conf/systemd.service [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 After=network.target
 
 [Service]
 Type=simple
 User=__APP__
 Group=__APP__
-WorkingDirectory=__FINALPATH__/
-ExecStart=__FINALPATH__/venv/bin/python3 __FINALPATH__/index.py
+WorkingDirectory=__INSTALL_DIR__/
+ExecStart=__INSTALL_DIR__/venv/bin/python3 __INSTALL_DIR__/index.py
```
Comment (alexAubin) [inline conf/systemd.service:10]:
```suggestion
ExecStart=__INSTALL_DIR__/venv/bin/python3 __INSTALL_DIR__/app/index.py
```

Dunno how it was working before ...

**Comment by alexAubin** [comment]
I'm super confused ... The service complains that "No such file or directory", either for the python bin in the venv (but it exists) or the index.py (but it exists as well) so wtf ...

**Comment by alexAubin** [comment]
Turns out the current app is level 1 because install works but Backup/restore fails because of 502 with the same issue : 

```
Dec 28 21:41:41 systemd[3366]: 13ft.service: Failed to set up standard output: No such file or directory
Dec 28 21:41:41 systemd[3366]: 13ft.service: Failed at step STDOUT spawning /var/www/13ft/venv/bin/python3: No such file or directory
```

(or maybe that's because of the stdout thing)

File: conf/systemd.service (hunk 3 above)
Comment (lapineige) [reply conf/systemd.service]: It wasn't...

**Comment by lapineige** [comment]
Backup and restore are not a big deal : there is nothing to backup, a fresh install is good enough.

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Pull request merged**